### PR TITLE
Upgrade to python 3.12

### PR DESCRIPTION
**Pull request by rapidmod**
1. upgraded to python 3.12
2. added chardet to test for proper encodings
3. added variables to use persistent db (and not create it each time)
4. added variables to ignore errors to continue the script instead of failing
5. upgraded langchain models and src imports

### CODE REVIEW

File: code_indexer_loop/test_api.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import pytest
 
-from code_indexer_loop.code_splitter import (CodeSplitter,
+from src.code_splitter import (CodeSplitter,
```
Comment (ricklamers) [inline code_indexer_loop/test_api.py:5]: Why not use package name?

File: code_indexer_loop/utils.py [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import hashlib
 
 
+# changed "hash_md5" to "md5" to avoid shadowing the built-in "hashlib.md5"
+# rapidmod.io - 04/04/2024
 def hash_md5(filename):
-    hash_md5 = hashlib.md5()
+    md5 = hashlib.md5()
```
Comment (ricklamers) [inline code_indexer_loop/utils.py:7]: hash_md5 != hashlib.md5?

File: code_indexer_loop/test_api.py (hunk 1 above)
Comment (rapidmod) [reply code_indexer_loop/test_api.py:5]: This was a mistake and changed back to package name in final
See: [put back in appropriate directory, builds fails loading langchain schema](https://github.com/definitive-io/code-indexer-loop/pull/3/commits/64853006cab99ef782debbfbfc0a7f0d48d6e9f3)
[6485300](https://github.com/definitive-io/code-indexer-loop/pull/3/commits/64853006cab99ef782debbfbfc0a7f0d48d6e9f3)

File: code_indexer_loop/utils.py (hunk 2 above)
Comment (rapidmod) [reply code_indexer_loop/utils.py:7]: There is too many shadows, the name makes sense, the inherited un intended consequences do not make sense

**Comment by rapidmod** [comment]
There is 1 more problem
Your package is my first python project with MAKEFILE I don't feel comfortable upgrading the requirements in your project. I have the basic concept but I don't fully understand everything that's goes on when you use MAKEFILE or flit.  I do have the requirements.txt file.

Either way I did learn from your project and will be implementing some of what I learned in my own projects. I did the work to update it so you didn't have too as thank you, use it as you wish or if you want me to add or change something let me know

**Pull request closed without merging**